lib/spack/spack/util/spack_yaml.py

```diff
@@ -70,25 +72,20 @@ def syaml_type(obj):
     return obj
 
 
-def markable(obj):
-    """Whether an object can be marked."""
-    return type(obj) in syaml_types
-
-
 def mark(obj, node):
     """Add start and end markers to an object."""
-    if not markable(obj):
-        return
-
-    if hasattr(node, 'start_mark'):
-        obj._start_mark = node.start_mark
-    elif hasattr(node, '_start_mark'):
-        obj._start_mark = node._start_mark
+    try:
+        if hasattr(node, 'start_mark'):
+            obj._start_mark = node.start_mark
+        elif hasattr(node, '_start_mark'):
+            obj._start_mark = node._start_mark
 
-    if hasattr(node, 'end_mark'):
-        obj._end_mark = node.end_mark
-    elif hasattr(node, '_end_mark'):
-        obj._end_mark = node._end_mark
+        if hasattr(node, 'end_mark'):
+            obj._end_mark = node.end_mark
+        elif hasattr(node, '_end_mark'):
+            obj._end_mark = node._end_mark
+    except AttributeError:
+        tty.debug("Failing type: " + str(type(obj)))
```

Comment:
This was a temporary message and I have since removed the try-except. I have now expanded the set of markable types (which now includes `ruamel.yaml` types like `CommentedMap`) since I feel it is important to keep track of what we are actually marking vs. a best-effort approach.